docs/blocks/index.md

```diff
@@ -1,51 +1,169 @@
-- `friction` - Controls how slippery the block is. For reference, ice has a slipperiness of 0.98.
+So for example, a simple implementation would look something like this:
 
-All these methods are *chainable* which means you can call them in series. See the `Blocks` class for examples of this.
+```java
+public static final RegistryObject<Block> MY_BETTER_BLOCK = BLOCKS.register("my_better_block", () -> new Block(BlockBehaviour.Properties.of()
+        .destroyTime(2.0f)
+        .explosionResistance(10.0f)
+        .sound(SoundType.GRAVEL)
+        .lightLevel(state -> 7)));
+```
+
+For further documentation, see the [`BlockBehaviour.Properties` reference][propertiesreference]. For more examples, or to look at the values used by Minecraft, have a look at the `Blocks` class.
 
 :::note
-Blocks have no setter for their `CreativeModeTab`. This is handled by the [`BuildCreativeModeTabContentsEvent`][creativetabs] if the block has an associated item (e.g. `BlockItem`). Furthermore, there is no setter for translation key of the block as it is generated from the registry name via `Block#getDescriptionId`.
+It is important to understand that a block in the world is not the same thing as in an inventory. What looks like a block in an inventory is actually a `BlockItem`, a special type of [item] that places a block when used. This also means that things like the creative tab or the max stack size are handled by the corresponding `BlockItem`.
+
+A `BlockItem` must be registered separately from the block. This is because a block does not necessarily need an item, for example if it is not meant to be collected (as is the case with fire, for example).
 :::
 
-### Advanced Blocks
+### More Functionality
 
-Of course, the above only allows for extremely basic blocks. If you want to add functionality, like player interaction, a custom class is required. However, the `Block` class has many methods and unfortunately not every single one can be documented here. See the rest of the pages in this section for things you can do with blocks.
+Directly using `Block` only allows for very basic blocks. If you want to add functionality, like player interaction or a non-1x1x1 hitbox, a custom class that extends `Block` is required. The `Block` class has many methods that can be overridden to do different things; these are documented in the [`Block` reference][blockreference]. See also the Using blocks section below for some of the most common use cases for blocks.
 
-Registering a Block
--------------------
+If you want to make a block that has different variants (think a slab that has a bottom, top, and double variant), you should use [blockstates]. And finally, if you want a block that stores additional data (think a chest that stores its inventory), a [block entity][blockentities] should be used. The rule of thumb here is that if you have a finite amount of states, use blockstates, and if you have an infinite or near-infinite amount of states, use a block entity.
```

Comment:
Again, this brings into an issue of how many states. Imagine if I had a slab which could be rotated in 24 different directions and had a bottom, top, and double. That is 72 states, which is all stored in memory. There comes a point where multiple finite amount of states will overflow the memory buffer.

Reply:
Do we have a definitive answer for this though? I'm not sure what the generally agreed-upon answer is here.

Reply:
There's not a general answer for this since it varies depending on the complexity of the block. My general sense is that if you need more than 8-9 bits of data (256 to 512 states), then it should be a block entity.

Reply:
Added the rough estimate.